### PR TITLE
Loader for D1B instrument

### LINKED ISSUE #29848: Create a loader for D1B
ILL's D1B needs a loader, as a first step to reduce its data.

### PR DESCRIPTION
**Pull request by MathieuTillet**
**Description of work.**
This PR adds a loader for ILL's instrument D1B.
Since the 2theta offset is not yet included in the nexus files, a field is added so the user can provide it himself.

**To test:**
- Load `External/Testing/Data/UnitTest/ILL/D1B/473432.nxs`.
- Check there is data, and that the instrument looks fine.

Fixes #29848 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by darominski — APPROVED** [review]
Looks good, data seems to be loaded correctly, along with the logs and the instrument view.

File: Framework/DataHandling/inc/MantidDataHandling/LoadILLDiffraction.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ class MANTID_DATAHANDLING_DLL LoadILLDiffraction
   void fillDataScanMetaData(const NeXus::NXDouble &);
   void fillMovingInstrumentScan(const NeXus::NXUInt &, const NeXus::NXDouble &);
   void fillStaticInstrumentScan(const NeXus::NXUInt &, const NeXus::NXDouble &,
-                                const NeXus::NXFloat &);
+                                const double &);
```
Comment (gvardany) [inline Framework/DataHandling/inc/MantidDataHandling/LoadILLDiffraction.h:67]: It does not make much sense to pass primitive scalar by reference, can you pass by value instead?

File: Framework/DataHandling/src/LoadILLDiffraction.cpp [hunk 2]
```diff
@@ -111,6 +113,8 @@ void LoadILLDiffraction::init() {
                   "Select the type of data, with or without calibration "
                   "already applied. If Auto then the calibrated data is "
                   "loaded if available, otherwise the raw data is loaded.");
+  declareProperty("TwoThetaOffset", 0.0,
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadILLDiffraction.cpp:116]: New properties should be declared to the end, could you move this?

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderILLParameterScan.py [hunk 3]
```diff
@@ -38,10 +38,10 @@ def category(self):
         return "ILL\\Diffraction;Diffraction\\Reduction"
 
     def summary(self):
-        return 'Performs powder diffraction data reduction for ILL instrument D20.'
+        return 'Performs powder diffraction data reduction for ILL instrument D20 and D1B.'
```
Comment (gvardany) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderILLParameterScan.py:41]: Would be nice to add a test for this with a couple of D1B runs.

**Pull request merged**